### PR TITLE
Release pasteup master

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

prevents you from releasing a new version of pasteup if you're:
- not on master
- have a dirty branch
## What is the value of this and can you measure success?

we need to make sure that released versions are actually code we have in master, not just something you've done and then change/delete etc.
## 

_Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?_

### CODE REVIEW

File: static/src/stylesheets/pasteup/publish.js [hunk 1]
```diff
@@ -12,10 +12,34 @@ var inquirer = require('inquirer');
...
+    return new Promise(function (resolve, reject) {
+        git.branch(function (branch) {
+            if (branch === 'master') {
+                resolve();
+            } else {
+                reject(new Error('notmaster'));
+            }
+        })
+    });
+}
```
Comment (OliverJAsh) [inline static/src/stylesheets/pasteup/publish.js]: Minor comment but, given the name, I would expect this function to return a boolean

File: static/src/stylesheets/pasteup/publish.js [hunk 2]
```diff
@@ -12,10 +12,34 @@ var inquirer = require('inquirer');
...
+function isClean() {
+    return isGitClean().then(function (isClean) {
+        if (isClean) {
+            return true;
+        } else {
+            throw new Error('notclean');
+        }
+    })
+}
+
```
Comment (OliverJAsh) [inline static/src/stylesheets/pasteup/publish.js]: Same here

File: static/src/stylesheets/pasteup/publish.js (hunk 1 above)
Comment (sndrs) [inline static/src/stylesheets/pasteup/publish.js]: yeah good point

**Comment by johnduffell** [comment]
looks like a great idea! 👍 for the principle and the code looks good from my limited knowledge

**Pull request merged**